### PR TITLE
More detailed HACKING.md and cleanup

### PR DESCRIPTION
**Pull request by galgalesh**
* [Explains](https://github.com/juju/charm-helpers/pull/3#issuecomment-330854113)
   - how to point override to local directory,
   - version bump during testing,
   - wheelhouse.txt format.

* Put README.test content in HACKING.md.

* Fixed `HACKING` -> `HACKING.md`

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
+2) Create an override file `override-wheelhouse.txt` that points to your own
+charmhelpers branch. *The format of this file is the same as pip's
+[`requirements.txt`](https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format)
+file.
 
-    -e git+https://github.com/<myuser>/charm-helpers#egg=charmhelpers
+    # Override charmhelpers by the version found in folder
+    -e /path/to/charmhelpers
+    # Or point it to a github repo with
+    # -e git+https://github.com/<myuser>/charm-helpers#egg=charmhelpers
```
Comment (johnsca) [inline HACKING.md]: The formatting is off on these.  I recommend:

```
# Override charmhelpers by the version found in folder
`-e /path/to/charmhelpers`

# Or point it to a github repo with
`-e git+https://github.com/<myuser>/charm-helpers#egg=charmhelpers`
```

# Override charmhelpers by the version found in folder
`-e /path/to/charmhelpers`

# Or point it to a github repo with
`-e git+https://github.com/<myuser>/charm-helpers#egg=charmhelpers`

File: HACKING.md [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
 
-Build the charm specifying the override file. *You might need to install the
+3) Build the charm specifying the override file. *You might need to install the
 candidate channel of the charm snap*
 
     charm build <mycharm> -w wheelhouse-overrides.txt
```
Comment (johnsca) [inline HACKING.md]: I recommend wrapping this with ``` to denote it as code:

```
charm build <mycharm> -w wheelhouse-overrides.txt
```

Comment (marcoceppi) [reply HACKING.md]: 4 line indent makes it a code block too though

Comment (marcoceppi) [reply HACKING.md]: Sorry, it's messed up because of the bullet, either way, 4 more indents or tick fence

Comment (galgalesh) [reply HACKING.md]: Good catch! It looked fine in the markdown viewer of atom. I'll fix it.

**Pull request merged**